### PR TITLE
feat: Validate newsfeed post and comment with zod

### LINKED ISSUE #214: No validation on newsfeed post and comment
### Issue Description

Add a zod validation like we have when you make a community post to the news feed post and comment.

1. There should be a minimum of 5 characters before accepted
2. Show error below input if validation fails

### Steps to Reproduce

_No response_

### Expected Behavior

_No response_

### Current Behavior

_No response_

### Screenshots (optional)

_No response_

### Operating System [e.g., Windows 10]

_No response_

### Browser [e.g., Chrome, Firefox]

_No response_

### Environment (locally, production, selfhosted)

_No response_

### Please check the boxes that apply to this issue report.

- [ ] I have searched the existing issues to make sure this is not a duplicate.
- [ ] I have provided steps to reproduce the issue.
- [ ] I have included relevant environment information.
- [ ] I have included any relevant screenshots.
- [ ] I understand that this is a voluntary contribution and that there is no guarantee of resolution.
- [ ] I want to work on creating a PR for this issue if approved

### PR DESCRIPTION
**Pull request by Chifez**
## What does this PR do?
This PR validates the post or comment being sent in the newsfeed using zod 

Fixes #214

https://github.com/rotimi-best/classroomio/assets/69228399/194fe4f8-5d25-404b-9ab4-b7a019caa6f1

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- run it on localhost

### Required

- [x] Filled out the "How to test" section in this PR
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the ClassroomIO Docs if changes were necessary

### CODE REVIEW

File: apps/dashboard/src/lib/utils/functions/validator.js [hunk 1]
```diff
@@ -4,9 +4,13 @@ import { validateEmail } from './validateEmail';
 
 const getOrgNameValidation = () =>
   z.string().min(5, { message: 'Organization must contain 5 or more characters' });
+
 const getSiteNameValidation = () =>
   z.string().min(5, { message: 'Site name must contain 5 or more characters' });
 
+const getNewsfeedValidation = () =>
+  z.string().min(5, { message: 'input must contain 5 or more characters' });
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/utils/functions/validator.js]: Please change text to:

```
Field must contain 5 or more characters
```

**Review by rotimi-best — CHANGES_REQUESTED** [review]
Looks good in general, just a minor change

File: apps/dashboard/src/lib/utils/functions/validator.js (hunk 1 above)
Comment (Chifez) [reply apps/dashboard/src/lib/utils/functions/validator.js]: alright

**Review by rotimi-best — APPROVED** [review]
LGTM, thanks

**Pull request merged**